setup.py

```diff
@@ -74,11 +74,32 @@ def add_ac(k, v):
         pass
 
 
+def rescue_docarray():
+    """Upgrading from 2.x to 3.x is broken (https://github.com/jina-ai/jina/issues/4194)
+    This function checks if docarray is broken and if so attempts to rescue it
+    """
+    try:
+        import docarray as docarray
+
+        __docarray_version__ = docarray.__version__
+
+    except AttributeError:
+        # Being here means docarray is not installed correctly, attempt to reinstall it
+        # as recommended by pip https://pip.pypa.io/en/latest/user_guide/#using-pip-from-your-program
+        import subprocess
+
+        subprocess.check_call(
```

Comment:
doesnt pip have a python package? This for instance would not work on my environment becauee I use pip3.7 for instance

Reply:
@numb3r3 could you throw some light here? i think you did something similar in `hub.uses(from_source)`

Reply:
I did this specifically because Pip says so: https://pip.pypa.io/en/latest/user_guide/#using-pip-from-your-program

Reply:
same method is utilized in `hub.uses(...)`. `[sys.executorable, '-m', 'pip']` always use the executable pip associated with the current python version you are using.